sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/AggregateBenchmark.scala

```diff
@@ -34,621 +34,508 @@ import org.apache.spark.unsafe.map.BytesToBytesMap
+        f()
+      }
 
-    max function bytecode size:              Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
-    ------------------------------------------------------------------------------------------------
-    codegen = F                                    709 /  803          0.9        1082.1       1.0X
-    codegen = T hugeMethodLimit = 10000           3485 / 3548          0.2        5317.7       0.2X
-    codegen = T hugeMethodLimit = 1500             636 /  701          1.0         969.9       1.1X
-     */
-  }
+      benchmark.run()
+    }
+
+    runBenchmark("max function bytecode size of wholestagecodegen") {
+      val N = 20 << 15
+
+      val benchmark = new Benchmark("max function bytecode size", N, output = output)
+
+      def f(): Unit = spark.range(N)
+        .selectExpr(
+          "id",
+          "(id & 1023) as k1",
+          "cast(id & 1023 as double) as k2",
+          "cast(id & 1023 as int) as k3",
+          "case when id > 100 and id <= 200 then 1 else 0 end as v1",
+          "case when id > 200 and id <= 300 then 1 else 0 end as v2",
+          "case when id > 300 and id <= 400 then 1 else 0 end as v3",
+          "case when id > 400 and id <= 500 then 1 else 0 end as v4",
+          "case when id > 500 and id <= 600 then 1 else 0 end as v5",
+          "case when id > 600 and id <= 700 then 1 else 0 end as v6",
+          "case when id > 700 and id <= 800 then 1 else 0 end as v7",
+          "case when id > 800 and id <= 900 then 1 else 0 end as v8",
+          "case when id > 900 and id <= 1000 then 1 else 0 end as v9",
+          "case when id > 1000 and id <= 1100 then 1 else 0 end as v10",
+          "case when id > 1100 and id <= 1200 then 1 else 0 end as v11",
+          "case when id > 1200 and id <= 1300 then 1 else 0 end as v12",
+          "case when id > 1300 and id <= 1400 then 1 else 0 end as v13",
+          "case when id > 1400 and id <= 1500 then 1 else 0 end as v14",
+          "case when id > 1500 and id <= 1600 then 1 else 0 end as v15",
+          "case when id > 1600 and id <= 1700 then 1 else 0 end as v16",
+          "case when id > 1700 and id <= 1800 then 1 else 0 end as v17",
+          "case when id > 1800 and id <= 1900 then 1 else 0 end as v18")
+        .groupBy("k1", "k2", "k3")
+        .sum()
+        .collect()
+
+      benchmark.addCase("codegen = F") { iter =>
+        spark.conf.set(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key, "false")
+        f()
+      }
 
+      benchmark.addCase("codegen = T hugeMethodLimit = 10000") { iter =>
+        spark.conf.set(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key, "true")
+        spark.conf.set(SQLConf.WHOLESTAGE_HUGE_METHOD_LIMIT.key, "10000")
+        f()
+      }
 
-  ignore("cube") {
-    val N = 5 << 20
+      benchmark.addCase("codegen = T hugeMethodLimit = 1500") { iter =>
+        spark.conf.set(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key, "true")
+        spark.conf.set(SQLConf.WHOLESTAGE_HUGE_METHOD_LIMIT.key, "1500")
```

Comment:
Although this is not a problem of this refactoring, this test suite seems to be unhealthy because the configuration from the previous benchmark is propagated to the next benchmark.

Can we fix this test suite to use `withSQLConf`?